aws-android-sdk-s3/src/main/java/com/amazonaws/mobileconnectors/s3/transferutility/TransferService.java

```diff
@@ -149,11 +149,8 @@ public void onReceive(Context context, Intent intent) {
                 new Thread(new Runnable() {
                     @Override
                     public void run() {
-                        if (networkConnected) {
```

Comment:
This needs to be worked on. The main feature of this class is to check for network connectivity and pause when offline and resume when online.

Reply:
I moved the network checking logic into the `pauseAllForNetwork `and `checkTransfersOnNetworkReconnect `methods so that each transfer can check if its network type is active or not.  Does that not work with the updated SDK code?

Reply:
@nasdf Thank you for the information. However my concern is that, when a developer passes the `TransferUtilityOptions` via `TransferUtility` object, it is used for the transfers (upload and download). However the NetworkInfoReceiver / TransferService is not aware of that connection type chosen by the developer and we are checking for ANY.

My suggestion would be to find a way to be consistent with the connection type and use that everywhere in the SDK (TransferUtility, Upload/DownloadTask, TransferService and TransferNetworkLossHandler).

I am working on this suggestion and discussing with other folks in the team. I would like to know your approach for this use-case as well.